cunumeric/module.py

```diff
@@ -2381,6 +2382,61 @@ def indices(
         return res_array
 
 
+def mask_indices(n: int, mask_func, k: int = 0) -> tuple[ndarray, ...]:
+    """
+    Return the indices to access (n, n) arrays, given a masking function.
+
+    Assume `mask_func` is a function that, for a square array a of size
+    ``(n, n)`` with a possible offset argument `k`, when called as
+    ``mask_func(a, k)`` returns a new array with zeros in certain locations
+    (functions like `triu` or `tril` do precisely this). Then this function
+    returns the indices where the non-zero values would be located.
+
+    Parameters
+    ----------
+    n : int
+        The returned indices will be valid to access arrays of shape (n, n).
+    mask_func : callable
+        A function whose call signature is similar to that of `triu`, `tril`.
```

Comment:
Please use
```
:func:`cunumeric:triu` etc.
```
so we get hyperlinking between docs. Similar to other instance of `tril` and `triu` within this docstring.